lib/gooddata/rest/connection.rb

```diff
@@ -466,6 +466,17 @@ def generate_request_id
         "#{@execution_id}:#{session_id}:#{call_id}"
       end
 
+      def enrich_error_message(exception)
+        begin
+          response = JSON.parse(exception.response.body, symbolize_names: true)
+          exception.define_singleton_method :to_s do
+            super() + ': ' + response[:error][:message] % response[:error][:parameters] + 'request_id: ' + response[:error][:requestId]
+          end
+        rescue JSON::ParserError
```

Comment:
swallowed exception

Reply:
What do you propose should be done in this case?

The current implementation will just leave the exception as is - does not enrich anything. Since many possible requests could be done that do not require any enriching (http response type etc.), I do not think logging an error or warning is a good idea.

Reply:
You're right. The original exception won't be swallowed so it's fine. ✔️